common/core/src/retry_policy.ts

```diff
@@ -0,0 +1,179 @@
+   *
+   * @param {Error} error The error encountered by the operation.
+   * @returns {boolean}   Whether the operation should be retried or not.
+   */
+  /*Codes_SRS_NODE_COMMON_RETRY_POLICY_16_001: [Any implementation of the `RetryPolicy` interface shall have a `shouldRetry` method used to evaluate if an error is "retryable" or not.]*/
+  shouldRetry: (error: Error) => boolean;
+}
+
+/**
+ * Parameters used with the {@link azure-iot-common.ExponentialBackOffWithJitter} policy to compute retry intervals.
+ */
+/*SRS_NODE_COMMON_RETRY_POLICY_16_009: [The default constants to use with the Math formula for the normal conditions retry are:
+```
+c = 100
+cMin = 100
+cMax = 10000
+ju = 0.25
+jd = 0.5
+```]*/
+export class ExponentialBackoffWithJitterParameters {
+  /**
+   * Initial retry interval: 100 ms by default.
+   */
+  c: number = 100;
+  /**
+   * Minimal interval between each retry. 100 milliseconds by default
+   */
+  cMin: number = 100;
+  /**
+   * Maximum interval between each retry. 10 seconds by default
+   */
+  cMax: number = 10000;
+  /**
+   * Jitter up factor. 0.25 by default.
+   */
+  ju: number = 0.25;
+  /**
+   * Jitter down factor. 0.5 by default
+   */
+  jd: number = 0.5;
+}
+
+/**
+ * Implements an Exponential Backoff with Jitter retry strategy.
+ * The function to calculate the next interval is the following (x is the xth retry):
+ * F(x) = min(Cmin+ (2^(x-1)-1) * rand(C * (1 – Jd), C*(1-Ju)), Cmax)
+ * @implements {RetryStrategy}
+ */
+/*Codes_SRS_NODE_COMMON_RETRY_POLICY_16_005: [The `ExponentialBackoffWithJitter` class shall implement the `RetryPolicy` interface.]*/
+export class ExponentialBackOffWithJitter implements RetryPolicy {
+  /**
+   * Retry parameters used to calculate the delay between each retry in normal situations (ie. not throttled).
+   */
+  normalParameters: ExponentialBackoffWithJitterParameters;
+  /**
+   * Retry parameters used to calculate the delay between each retry in throttled situations.
+   */
+  throttledParameters: ExponentialBackoffWithJitterParameters;
+  /**
+   * Boolean indicating whether the first retry should be immediate (if set to true) or after the normalParameters.c delay (if set to false).
+   */
+  immediateFirstRetry: boolean;
+
+  private _errorFilter: ErrorFilter;
+
+  /**
+   * Initializes a new instance of the {@link azure-iot-common.ExponentialBackOffWithJitter} class.
+   * @param maxTimeout            maximum allowed timeout in milliseconds.
+   * @param immediateFirstRetry   boolean indicating whether the first retry should be immediate (default) or wait the first interval (c value).
+   */
+  constructor(immediateFirstRetry?: boolean, errorFilter?: ErrorFilter, ) {
+    this._errorFilter = errorFilter ? errorFilter : new DefaultErrorFilter();
+    this.immediateFirstRetry = immediateFirstRetry !== false ? true : false;    // should default to true if not specified.
```

Comment:
>true : false [](start = 63, length = 12)

Needed?

Reply:
its a bit awkward but the idea is that it should be true by default, unless it's specifically set to false. so "undefined", "null" or any other value really should count as true.

Reply:
shall be corrected with `immediateFirstRetry !== false` (and remove the ?: expression)